manifests/init.pp

```diff
@@ -88,17 +91,29 @@
 #   Whether or not the module should manage the service.
 #
 # @param service_name
-#   Customise the name of the system service
+#   Customise the name of the system service.
 #
 # @param service_ensure
-#   Whether the service should be running or stopped
+#   Whether the service should be running or stopped.
 #
 # @param service_enable
-#   Whether the service should be enabled or disabled
+#   Whether the service should be enabled or disabled.
+#
+# @param manage_repo
+#   Whether the APT/YUM(COPR) repository should be installed. Only relevant when $install_method is 'repo'.
+#
+# @param distribution_channel
+#   Whether to use stable or testing distribution channel. Only relevant on Debian family and only when $install_method is 'repo'.
+#
+# @param package_name
+#   Name of the caddy package to use. Only relevant when $install_method is 'repo'.
+#
+# @param package_ensure
+#   Whether to install or remove the caddy package. Only relevant when $install_method is 'repo'.
 #
 class caddy (
   String[1]                      $version                         = '2.0.0',
-  Optional[Enum['github']]       $install_method                  = undef,
+  Optional[Enum['github','repo']] $install_method                 = undef,
```

Comment:
I am wondering if we should remove the 'Optional' part and add something like 'upstream'. Some people use Hiera to change settings in modules, and the current way make it impossible to revert to upstream on one host if another level in the hierarchy set it to something else.

Reply:
Yes, I have that in my mind, but it'll be breaking change (potentially, if someone do `class { 'caddy': install_method => undef }`). So far I'm trying to keep backward compatibility.

Reply:
I mean, I'll change this, just in different PR.